src/net_processing.cpp

```diff
@@ -155,6 +155,9 @@ static constexpr uint32_t MAX_GETCFHEADERS_SIZE = 2000;
 static constexpr size_t MAX_PCT_ADDR_TO_SEND = 23;
 /** The maximum number of address records permitted in an ADDR message. */
 static constexpr size_t MAX_ADDR_TO_SEND{1000};
+/** The maximum rate of address records we're willing to process on average. Can be bypassed using
+ *  the NetPermissionFlags::Addr permission. */
+static constexpr double MAX_ADDR_RATE_PER_SECOND{0.1};
```

Comment:
For bucket size we use `MAX_ADDR_TO_SEND` in the code, but they are semantically different. Maybe add new constant `ADDR_RATE_BUCKET_SIZE`? Should that be a little bit higher than `MAX_ADDR_TO_SEND`? For example:

1. The peer sends us some unsolicited addr
2. We send getaddr and fill the bucket to the max 1000
3. We receive the unsolicited addr from 1.
4. We receive the getaddr response from 2 (1000 addresses)

=> some address is going to be dropped unnecessary, even though the peer did not "spam" us.

Reply:
2 is incorrect. We increment the bucket by 1000, not set it to 1000.

Reply:
Right! It is capped to `1000`, but when expecting getaddr response we just do `+= 1000` without checking. Sorry for the noise.

Reply:
I've introduced a separate `MAX_ADDR_PROCESSING_TOKEN_BUCKET` constant, also initialized to 0. Also amended a few comments to be clearer about the fact that GETADDR can exceed this limit.